code/datums/traits/negative.dm

```diff
@@ -370,3 +370,108 @@
+/datum/quirk/junkie
+	name = "Junkie"
+	desc = "You can't get enough of hard drugs."
+	value = -2
+	var/drug_list = list("crank", "krokodil", "morphine", "happiness", "methamphetamine") //List of possible IDs
+	var/reagent_id //ID picked from list
+	var/datum/reagent/reagent_type //If this is defined, reagent_id will be unused and the defined reagent type will be instead.
+	var/datum/reagent/R
+	gain_text = "<span class='danger'>You suddenly feel the craving for drugs.</span>"
+	lose_text = "<span class='notice'>You feel like you should kick your drug habit.</span>"
+	medical_record_text = "Patient has a history of hard drugs."
+	var/where_drug
+	var/obj/item/drug_container_type //If this is defined before pill generation will be skipped, this is the type of the pill bottle.
+	var/obj/item/D
+	var/where_accessory
+	var/obj/item/accessory_type //If this is null, it won't be spawned.
+	var/obj/item/A
+	var/tick_counter = 0
+
+/datum/quirk/junkie/on_spawn()
+	var/mob/living/carbon/human/H = quirk_holder
+	reagent_id = pick(drug_list)
+	if (!reagent_type)
+		reagent_type = GLOB.chemical_reagents_list[reagent_id]
+	R = new reagent_type(null)
+	H.reagents.addiction_list.Add(R)
+	if (!drug_container_type)
+		drug_container_type = /obj/item/storage/pill_bottle
+		D = new drug_container_type(get_turf(quirk_holder))
+		var/pill_state = "pill[rand(1,20)]"
+		for(var/i in 1 to 7)
+			var/obj/item/reagent_containers/pill/P = new /obj/item/reagent_containers/pill(D)
+			P.icon_state = pill_state
+			P.list_reagents = list(reagent_id = 0.5)
+	D = new drug_container_type(get_turf(quirk_holder))
+	if (accessory_type)
+		A = new accessory_type(get_turf(quirk_holder))
+	var/list/slots = list(
+		"in your left pocket" = SLOT_L_STORE,
+		"in your right pocket" = SLOT_R_STORE,
+		"in your backpack" = SLOT_IN_BACKPACK
+	)
+	where_drug = H.equip_in_one_of_slots(D, slots, FALSE) || "at your feet"
+	if (A)
+		where_accessory = H.equip_in_one_of_slots(A, slots, FALSE) || "at your feet"
+
+/datum/quirk/junkie/post_add()
+	if(where_drug == "in your backpack" || where_accessory == "in your backpack")
+		var/mob/living/carbon/human/H = quirk_holder
+		SEND_SIGNAL(H.back, COMSIG_TRY_STORAGE_SHOW, H)
+	if (!accessory_type)
+		to_chat(quirk_holder, "<span class='boldnotice'>There is a [D.name] of [R.name] [where_drug]. Better hope you don't run out...</span>")
+
+/datum/quirk/junkie/on_process()
+	var/mob/living/carbon/human/H = quirk_holder
+	if (tick_counter == 60) //Halfassed optimization
+		var/in_list = FALSE
+		if (!isemptylist(H.reagents.addiction_list))
+			for (var/datum/reagent/entry in H.reagents.addiction_list)
+				if(istype(entry, reagent_type))
+					in_list = TRUE
+					break
+		if(!in_list)
+			H.reagents.addiction_list.Add(R)
+			R.addiction_stage = 0
+			to_chat(quirk_holder, "<span class='danger'>You thought you kicked it, but you suddenly feel like you need [R.name] again...")
+		tick_counter = 0
+	else
+		tick_counter = tick_counter + 1
```

Comment:
```suggestion
		++tick_counter
```

Reply:
I couldn't remember if dm had an increment operator and was too lazy to go check.